hoot-core/src/main/cpp/hoot/core/visitors/SetTagVisitor.cpp

```diff
@@ -66,19 +73,40 @@ void SetTagVisitor::setConfiguration(const Settings& conf)
   {
     _elementType = ElementType::fromString(configOptions.getSetTagVisitorElementType());
   }
+  _overwriteExistingTag = configOptions.getSetTagVisitorOverwrite();
+
+  LOG_VART(_k);
+  LOG_VART(_v);
+  LOG_VART(_appendToExistingValue);
+  LOG_VART(_elementType);
+  LOG_VART(_overwriteExistingTag);
 }
 
 void SetTagVisitor::_setTag(const ElementPtr& e, QString k, QString v)
 {
+  if (k.isEmpty())
+  {
+    throw IllegalArgumentException("You must set the key in the SetTagVisitor class.");
+  }
+  if (v.isEmpty())
+  {
+    throw IllegalArgumentException("You must set the value in the SetTagVisitor class.");
+  }
+
+  LOG_VART(e->getElementId());
+
+  LOG_VART(e->getElementType());
   if (_elementType != ElementType::Unknown && e->getElementType() != _elementType)
   {
     return;
   }
 
-  if (k.isEmpty())
+  LOG_VART(e->getTags().contains(k));
+  if (!_overwriteExistingTag && e->getTags().contains(k))
   {
-    throw IllegalArgumentException("You must set the key in the SetTagVisitor class.");
+    return;
   }
```

Comment:
made it so you can have an option to not overwrite existing tags in SetTagVisitor